lib/spack/docs/packaging_guide_creation.rst

```diff
@@ -777,28 +782,39 @@ The logic behind this sort order is two-fold:
 #. The most-recent development version of a package will usually be newer than any released numeric versions.
    This allows the ``@develop`` version to satisfy dependencies like ``depends_on(abc, when="@x.y.z:")``
 
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
-Specifying versions in other directives
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-Many Spack directives accept versions speciers, for example
+.. _version_constraints:
+
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Specifying version constraints
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Many Spack directives allow limiting versions to support features such as :ref:`version_compatibility`.
+These constraints on package specs are defined using the ``@<specifier>`` syntax (see :ref:`version-specifier` for more information).
+
+For example,
 
 .. code-block:: python
 
+   depends_on("foo")
    depends_on("python@3")
-   conflicts("foo@1.2.3:", when="@:4.5")
 
-When specifying versions in Spack using the ``@<specifier>`` syntax, you can use either ranges or specific versions.
-It is generally recommended to use ranges instead of specific versions when packaging to avoid overly constraining dependencies, patches, and conflicts.
+   conflicts("^foo@1.2.3:", when="@:4.5")
+
+illustrates three forms of a version range constraint.
+First, the package depends on *any* ``python`` with ``3`` as the major version number (e.g., ``3.13.5``).
+Then there is a conflict with the package ``foo`` dependency *at version ``1.2.3`` or newer*, which is triggered for builds of the package at *any version up to and including version ``4.5``*.
```

Comment:
rst does not supported nested formatting with asterisk & backticks :(

Reply:
Darn. I forgot to check that when I was looking at the output. Thanks for letting me know.

Reply:
Darn. I forgot to check that when I was looking at the output. Thanks for letting me know.

Reply:
Are you sure the version that Spack is using doesn't?

<img width="739" height="122" alt="Screenshot 2025-07-24 at 9 37 32 AM" src="https://github.com/user-attachments/assets/86aa7d30-65a8-4ed2-ad06-e622c1768c9c" />